Sources/PostgresNIO/Data/PostgresData+JSONB.swift

```diff
@@ -0,0 +1,60 @@
+import Foundation
+
+fileprivate let jsonBVersionBytes: [UInt8] = [0x01]
+
+extension PostgresData {
+    public init(jsonb jsonData: Data) {
+        let jsonBDataBytes = [UInt8](jsonData)
+        
+        var buffer = ByteBufferAllocator().buffer(capacity: jsonBVersionBytes.count + jsonBDataBytes.count)
+        buffer.writeBytes(jsonBVersionBytes)
+        buffer.writeBytes(jsonBDataBytes)
+        
+        self.init(type: .jsonb, formatCode: .binary, value: buffer)
+    }
+    
+    public var jsonb: Data? {
+        guard var value = self.value else {
+            return nil
+        }
+
+        guard let versionBytes = value.readBytes(length: jsonBVersionBytes.count), [UInt8](versionBytes) == jsonBVersionBytes else {
+            return nil
+        }
+        
+        guard let dataBytes = value.readBytes(length: value.readableBytes) else {
+            return nil
+        }
+        
+        return Data(dataBytes)
+    }
+}
+
+public protocol JSONBCodable: Codable, PostgresDataConvertible {
```

Comment:
Instead of a protocol, could we add this functionality as new methods on `PostgresData`? Something like:

```swift
struct Foo: Codable { ... }

let someFoo: Foo = ...
let data = try PostgresData(jsonb: someFoo)

let someData: PostgresData = ...
let foo = try someData.jsonb(as: Foo.self)
```

This would allow us to preserve the errors thrown by JSONEncoder/Decoder which would be valuable.

Reply:
Hi, I made a small change, keep the errors in PostgresData convert.
And the protocol provides service for model. something like:

```swift
struct Node: Model {
    struct Meta: Codable, JSONBCodable {
        let foo: String
    }
    let meta = Field<Meta>("meta")
}
```

Reply:
Maybe we should name this `PostgresJSONBCodable` or `PostgresJSONCodable`? Calling it `JSONBCodable` seems a bit general for a low-level package.

Reply:
Hi, I have rename the protocol name to `PostgresJSONBCodable`.